test/collection_test.js

```diff
@@ -157,6 +159,45 @@ describe("Collection", () => {
       });
     });
 
+    describe("hooks registration", () => {
+      function registerHooks(hooks) {
+        return new Collection(TEST_BUCKET_NAME, TEST_COLLECTION_NAME, api, {
+          hooks,
+          adapter: IDB
```

Comment:
There are existing tests for async transformers [here](https://github.com/Kinto/kinto.js/blob/master/test/collection_test.js#L930-L944).